src/resources/shipment.js

```diff
@@ -130,14 +160,17 @@ export default (api) =>
 
     /**
      * Regenerate rates of a shipment.
+     * @param {boolean} carbonOffset
      * @returns {this}
      */
-    async regenerateRates() {
+    async regenerateRates(carbonOffset = false) {
```

Comment:
Thought: In the PRs I made, I called this `withCarbonOffset` to suggest that this was a boolean rather than an object or value the user needed to pass in. Can we get consensus @Justintime50 ?

Reply:
I agree, `withCarbonOffset` makes more sense to guide the user to know what is required for this param. Let's change it.